llama_index/embeddings/gradient.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from typing import Any, List, Optional
+
+from llama_index.bridge.pydantic import Field, PrivateAttr
+from llama_index.embeddings.base import BaseEmbedding, Embedding
+
+logger = logging.getLogger(__name__)
+
+
+# For bge models that Gradient AI provides, it is suggested to add the instruction for retrieval.
+# Reference: https://huggingface.co/BAAI/bge-large-en-v1.5#model-list
+QUERY_INSTRUCTION_FOR_RETRIEVAL = (
+    "Represent this sentence for searching relevant passages:"
+)
+
+
+class GradientEmbedding(BaseEmbedding):
+    """GradientAI embedding models.
+
+    This class provides an interface to generate embeddings using a model
+    deployed in Gradient AI. At the initialization it requires a model_id
+    of the model deployed in the cluster.
+
+    Note:
+        Requires `gradientai` package to be available in the PYTHONPATH. It can be installed with
+        `pip install gradientai`.
+    """
+
+    _gradient: Any = PrivateAttr()
+    _model: Any = PrivateAttr()
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "GradientEmbedding"
+
+    def __init__(
+        self,
```

Comment:
can is constructor be kwargs only

Reply:
IIUC if you don't make some parameters explicit they don't show up in the IntelliSense?

<img width="1258" alt="2023-10-09_22-44-34" src="https://github.com/Preemo-Inc/llama_index/assets/158472/153b0ac6-697b-475d-ac36-1048e4daa5ba">

<img width="1320" alt="2023-10-09_22-46-33" src="https://github.com/Preemo-Inc/llama_index/assets/158472/f6fb6129-28c4-43f7-ad41-d2fe7a6770ed">
 at least for me?

Reply:
Nah I mean like `(self, *, ...)`

Reply:
THAT's what it's for!

Reply:
done!